packages/legend-application-query/src/stores/QueryEditorStore.ts

```diff
@@ -525,6 +525,9 @@ export abstract class QueryEditorStore {
       this.queryBuilderState = (yield this.initializeQueryBuilderState(
         stopWatch,
       )) as QueryBuilderState;
+      this.queryBuilderState.changeDetectionState.initialize(
```

Comment:
why do we need this ?

Reply:
This is for making sure `queryBuilderState.changeDetectionState.hasChanged` is functioning 
without this, change detection doesn't detect there are changes

Reply:
and at what point are we comparing the query? 
does have `Class.all()` count as a change ?

Reply:
I don't think .all() should be counted as a change e.g. for the`creating a query from mapping` entry, the class has been automatically chosen.

Reply:
yeah i dont either, which is why i asked the question, where do we start the change detection

Reply:
since the query doesn't just contain .all()

 `check if the rawLambda is not stub` will mark .all() as a change

Reply:
My question was more on using change detection method. 
Would we consider .all() as a change using change detection.

Reply:
No, it only will mark there is a change if the query doesn't just contain .all()